cli/ballerina-cli/src/main/java/io/ballerina/cli/task/RunBuildToolsTask.java

```diff
@@ -0,0 +1,359 @@
+    private final boolean exitWhenFinish;
+    private ClassLoader toolClassLoader = this.getClass().getClassLoader();
+    ServiceLoader<CodeGeneratorTool> toolServiceLoader = ServiceLoader.load(CodeGeneratorTool.class, toolClassLoader);
+    private final Map<String, ToolContext> toolContextMap = new HashMap<>();
+
+    public RunBuildToolsTask(PrintStream out) {
+        this.outStream = out;
+        this.exitWhenFinish = true;
+    }
+
+    @Override
+    public void execute(Project project) {
+        project.setToolContextMap(toolContextMap);
+
+        // Print all build tool manifest diagnostics
+        Collection<Diagnostic> toolManifestDiagnostics = project.currentPackage().manifest().diagnostics()
+                .diagnostics().stream().filter(diagnostic -> diagnostic.diagnosticInfo().code()
+                        .startsWith(TOOL_DIAGNOSTIC_CODE_PREFIX)).toList();
+        toolManifestDiagnostics.forEach(outStream::println);
+
+        // Read the build tool entries specified the Ballerina.toml
+        List<Tool> toolEntries = project.currentPackage().manifest().tools();
+        if (toolEntries.isEmpty()) {
+            return;
+        }
+        this.outStream.println("\nExecuting Build Tools");
+
+        // Populate the tool context map
+        for (Tool toolEntry : toolEntries) {
+            // Populate tool context
+            ToolContext toolContext = ToolContext.from(toolEntry, project.currentPackage(), outStream);
+            toolContextMap.put(toolEntry.id(), toolContext);
+        }
+        BuildToolResolution buildToolResolution;
+        try {
+            buildToolResolution = project.currentPackage().getBuildToolResolution();
+        } catch (ProjectException e) {
+            CommandUtil.printError(this.outStream, e.getMessage(), null, false);
+            CommandUtil.exitError(this.exitWhenFinish);
+            return;
+        }
+        buildToolResolution.getDiagnosticList().forEach(outStream::println);
+        List<BuildTool> resolvedTools = buildToolResolution.getResolvedTools();
+        List<BuildTool> centralDeliveredResolvedTools = resolvedTools.stream().filter(tool -> !DEFAULT_VERSION
+                .equals(tool.version().toString())).toList();
+        if (!centralDeliveredResolvedTools.isEmpty()) {
+            if (!project.buildOptions().offlineBuild()) {
+                pullLocallyUnavailableTools(centralDeliveredResolvedTools, buildToolResolution);
+            }
+            toolClassLoader = createToolClassLoader(centralDeliveredResolvedTools);
+            toolServiceLoader = ServiceLoader.load(CodeGeneratorTool.class, toolClassLoader);
+        }
+
+        // We run only the entries of resolved tools.
+        List<PackageManifest.Tool> resolvedToolEntries = toolEntries.stream()
+                .filter(toolEntry -> resolvedTools.stream()
+                        .anyMatch(tool -> tool.id().value().equals(toolEntry.type().split("\\.")[0])))
+                .toList();
```

Comment:
```suggestion
        List<PackageManifest.Tool> resolvedToolEntries = toolEntries.stream()
                .filter(toolEntry -> resolvedTools.stream()
                .anyMatch(tool -> tool.id().value().equals(toolEntry.type().split("\\.")[0])))
                .toList();
```
(Minor formatting)

Reply:
I think this is the correct formatting since `anyMatch` happens within the `filter` function.